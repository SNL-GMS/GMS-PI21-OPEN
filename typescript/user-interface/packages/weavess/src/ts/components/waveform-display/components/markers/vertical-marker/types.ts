import type { WeavessTypes } from '@gms/weavess-core';

export interface VerticalMarkerProps {
  /** Color as a string */
  color: string;

  /** Line style as a LineStyle */
  lineStyle: WeavessTypes.LineStyle;

  /** Percentage Location 0-100 as a number */
  percentageLocation: number;

  /** react key but called name since key was causing fortify issues */
  name?: string;
}

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface VerticalMarkerState {}
