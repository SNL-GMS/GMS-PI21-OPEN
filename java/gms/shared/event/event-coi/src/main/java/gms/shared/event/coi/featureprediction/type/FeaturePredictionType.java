package gms.shared.event.coi.featureprediction.type;

import com.fasterxml.jackson.annotation.JsonValue;
import com.fasterxml.jackson.core.JsonParser;
import com.fasterxml.jackson.databind.DeserializationContext;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
import gms.shared.event.coi.featureprediction.value.ArrivalTimeFeaturePredictionValue;
import gms.shared.event.coi.featureprediction.value.FeaturePredictionValue;
import gms.shared.event.coi.featureprediction.value.NumericFeaturePredictionValue;

import java.io.IOException;
import java.util.Map;
import java.util.Optional;

/**
 * Class for an instantiable object that represents the type of feature prediction. Each "type"
 * is associated with a FeaturePredictionValue class; together, they fully define what each type of
 * feature prediction looks like.
 *
 * @param <T> The FeaturePredictionValue class that should be coupled with this type.
 */
@JsonDeserialize(using = FeaturePredictionType.Deserializer.class)
public abstract class FeaturePredictionType<T extends FeaturePredictionValue<?, ?, ?>> {

  public enum TypeNames {
    ARRIVAL_TIME,
    SLOWNESS,
    EMERGENCE_ANGLE,
    RECEIVER_TO_SOURCE_AZIMUTH,
    SOURCE_TO_RECEIVER_AZIMUTH,
    SOURCE_TO_RECEIVER_DISTANCE,
  }

  private final TypeNames name;

  private FeaturePredictionType(
    TypeNames name
  ) {
    this.name = name;
  }

  @JsonValue
  public TypeNames getName() {
    return name;
  }

  @Override
  public String toString() {
    return "FeaturePredictionType{" +
      "name='" + name + '\'' +
      '}';
  }

  public abstract Class<T> getTypeValueClass();

  //
  // Constants for the supported feature prediction types.
  //

  public static final FeaturePredictionType<ArrivalTimeFeaturePredictionValue> ARRIVAL_TIME_PREDICTION_TYPE =
    new FeaturePredictionType<>(TypeNames.ARRIVAL_TIME) {
      @Override
      public Class<ArrivalTimeFeaturePredictionValue> getTypeValueClass() {
        return ArrivalTimeFeaturePredictionValue.class;
      }
    };

  public static final FeaturePredictionType<NumericFeaturePredictionValue> SLOWNESS_PREDICTION_TYPE =
    new FeaturePredictionType<>(TypeNames.SLOWNESS) {
      @Override
      public Class<NumericFeaturePredictionValue> getTypeValueClass() {
        return NumericFeaturePredictionValue.class;
      }
    };

  public static final FeaturePredictionType<NumericFeaturePredictionValue> EMERGENCE_ANGLE_PREDICTION_TYPE =
    new FeaturePredictionType<>(TypeNames.EMERGENCE_ANGLE) {
      @Override
      public Class<NumericFeaturePredictionValue> getTypeValueClass() {
        return NumericFeaturePredictionValue.class;
      }
    };

  public static final FeaturePredictionType<NumericFeaturePredictionValue> RECEIVER_TO_SOURCE_AZIMUTH_PREDICTION_TYPE =
    new FeaturePredictionType<>(TypeNames.RECEIVER_TO_SOURCE_AZIMUTH) {
      @Override
      public Class<NumericFeaturePredictionValue> getTypeValueClass() {
        return NumericFeaturePredictionValue.class;
      }
    };

  public static final FeaturePredictionType<NumericFeaturePredictionValue> SOURCE_TO_RECEIVER_AZIMUTH_PREDICTION_TYPE =
    new FeaturePredictionType<>(TypeNames.SOURCE_TO_RECEIVER_AZIMUTH) {
      @Override
      public Class<NumericFeaturePredictionValue> getTypeValueClass() {
        return NumericFeaturePredictionValue.class;
      }
    };

  public static final FeaturePredictionType<NumericFeaturePredictionValue> SOURCE_TO_RECEIVER_DISTANCE_PREDICTION_TYPE =
    new FeaturePredictionType<>(TypeNames.SOURCE_TO_RECEIVER_DISTANCE) {
      @Override
      public Class<NumericFeaturePredictionValue> getTypeValueClass() {
        return NumericFeaturePredictionValue.class;
      }
    };

  /**
   * Custom deserializer used by Jackson. The serialized JSON should contain a single field with
   * a single string that is the name of one of the types, and that string will be returned
   * be JsonParser::getValueAsString.
   */
  static class Deserializer extends StdDeserializer<FeaturePredictionType<?>> {

    private static final Map<TypeNames, FeaturePredictionType<?>> nameTypeMap = Map.of(
      TypeNames.ARRIVAL_TIME, ARRIVAL_TIME_PREDICTION_TYPE,
      TypeNames.SLOWNESS, SLOWNESS_PREDICTION_TYPE,
      TypeNames.EMERGENCE_ANGLE, EMERGENCE_ANGLE_PREDICTION_TYPE,
      TypeNames.RECEIVER_TO_SOURCE_AZIMUTH, RECEIVER_TO_SOURCE_AZIMUTH_PREDICTION_TYPE,
      TypeNames.SOURCE_TO_RECEIVER_AZIMUTH, SOURCE_TO_RECEIVER_AZIMUTH_PREDICTION_TYPE,
      TypeNames.SOURCE_TO_RECEIVER_DISTANCE, SOURCE_TO_RECEIVER_DISTANCE_PREDICTION_TYPE
    );

    public Deserializer() {
      this(null);
    }

    public Deserializer(Class<?> vc) {
      super(vc);
    }

    @Override
    public FeaturePredictionType<?> deserialize(JsonParser p, DeserializationContext ctxt)
      throws IOException {
      return Optional.ofNullable(nameTypeMap.get(TypeNames.valueOf(p.getValueAsString())))
        .orElseThrow();
    }
  }
}
