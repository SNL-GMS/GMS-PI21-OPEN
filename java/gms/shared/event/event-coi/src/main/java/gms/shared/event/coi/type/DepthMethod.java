package gms.shared.event.coi.type;

import com.google.common.base.Functions;
import com.google.common.base.Preconditions;
import com.google.common.collect.ImmutableMap;

import java.util.Arrays;

public enum DepthMethod {
  A("a"),
  F("f"),
  D("d"),
  R("r"),
  G("g"),
  UNKNOWN("-");


  private static final ImmutableMap<String, DepthMethod> LABELS_BY_MODEL;

  static {
    LABELS_BY_MODEL = Arrays.stream(values())
      .collect(ImmutableMap.toImmutableMap(DepthMethod::getLabel, Functions.identity()));
  }

  public final String label;


  DepthMethod(String label) {
    this.label = label;
  }

  public static DepthMethod valueOfLabel(String label) {
    Preconditions.checkNotNull(label, "label cannot be null");
    Preconditions.checkArgument(!label.isBlank(), "label cannot be blank");
    return LABELS_BY_MODEL.get(label);
  }

  public String getLabel() {
    return label;
  }

}

