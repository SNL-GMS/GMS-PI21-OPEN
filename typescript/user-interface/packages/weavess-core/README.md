# Weavess Util

Provides access to core WEAVESS features without a dependency on the window object. This allows these WEAVESS features to be used in web workers in external packages.

## Build Scripts
  * `clean:node_modules`: removes the node_modules directory
  * `clean:build`: removes all of the build directories, files, and artifacts
  * `clean`: cleans and removes all
  * `build:eslint`: runs eslint checks on the package for the source
  * `build:eslint:test`: runs eslint checks on the package for tests
  * `build`: runs the build of the package for the source
  * `build:test`: runs the build of the package for tests
  * `watch`: runs the build and watches for changes to recompile
  * `docs`: generates the package source documentation
  * `sonar`: runs sonar lint checks
  * `test`: builds and runs the package jest tests as development
  * `test:dev:jest`: runs the package jest tests as development
  * `test:prod:jest`: runs the package jest tests as production
  * `test:dev`: builds and runs the package jest tests as development
  * `test:prod`: builds and runs the package jest tests as production 
  * `version`: returns the version of the package

## Documentation

To generate HTML documentation files:

```bash
yarn docs
```

## Building

```bash
[.../weavess-core] $ yarn build
```

## Generating Binaries

```bash
[.../weavess-core] $ yarn build
```
