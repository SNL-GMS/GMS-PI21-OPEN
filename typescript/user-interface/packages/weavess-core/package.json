{
  "name": "@gms/weavess-core",
  "version": "0.1.0",
  "description": "WEAVESS Core",
  "license": "UNLICENSED",
  "author": "GMS Development Team",
  "sideEffects": false,
  "module": "./lib/weavess-core.js",
  "typings": "./lib/weavess-core.d.ts",
  "files": [
    "lib"
  ],
  "scripts": {
    "clean:node_modules": "yarn rimraf node_modules/",
    "clean:build": "yarn rimraf lib/ dist/ docs/ coverage/ bundle-analyzer/ artifacts/ .jest-cache/ .tsbuildinfo .tsbuildinfo-test .eslintcache .eslintcache-test junit.xml",
    "clean": "npm-run-all -s clean:build clean:node_modules",
    "build:eslint": "yarn eslint -c .eslintrc-config.yaml 'src/**/*.{ts,tsx}' --quiet --cache --cache-file .eslintcache",
    "build:eslint:test": "yarn eslint -c .eslintrc-config-test.yaml '{__tests__,__mocks__,__integration-tests__}/**/*.{ts,tsx}' --quiet --cache --cache-file .eslintcache-test",
    "build": "yarn tsc -b ./tsconfig-build.json",
    "build:test": "yarn tsc -b ./tsconfig-test.json",
    "watch": "yarn tsc -b ./tsconfig-build.json -watch",
    "docs": "yarn clean:docs && typedoc --out docs ./src/ --name weavess-core --mode file --readme README.md --target ES6 --includeDeclarations --excludeExternals --excludeNotExported --hideGenerator",
    "sonar": "echo \"run 'yarn sonar' from the top-level directory\"",
    "test": "yarn test:dev",
    "test:dev:jest": "jest --config jest.config.js --maxWorkers=50% --verbose --bail",
    "test:prod:jest": "jest --config jest.config.js --maxWorkers=4 --silent --bail --coverageReporters=lcovonly --reporters=default --reporters=jest-junit",
    "test:dev": "npm-run-all -s build:test test:dev:jest",
    "test:prod": "npm-run-all -s build:test test:prod:jest",
    "version": "echo $npm_package_version"
  },
  "dependencies": {
    "csv-parse": "4.8.2",
    "d3": "5.5.0",
    "immutable": "4.0.0-rc.12",
    "lodash": "4.17.21",
    "math-random": "2.0.1",
    "moment": "2.26.0",
    "react": "18.1.0",
    "react-dom": "18.1.0",
    "uuid": "3.3.3"
  },
  "devDependencies": {
    "@blueprintjs/eslint-plugin": "2.2.0",
    "@testing-library/jest-dom": "5.16.4",
    "@testing-library/react": "13.2.0",
    "@types/d3": "4.12.0",
    "@types/faker": "4.1.8",
    "@types/jest": "27.0.1",
    "@types/lodash": "4.14.149",
    "@types/react": "18.0.8",
    "@types/react-dom": "18.0.3",
    "@types/react-router-dom": "5.3.3",
    "@types/uuid": "3.4.6",
    "@typescript-eslint/eslint-plugin": "5.33.0",
    "@typescript-eslint/parser": "5.33.0",
    "copyfiles": "2.1.1",
    "eslint": "8.22.0",
    "eslint-config-airbnb": "19.0.4",
    "eslint-config-prettier": "8.5.0",
    "eslint-import-resolver-typescript": "3.4.1",
    "eslint-plugin-import": "2.26.0",
    "eslint-plugin-jest": "26.8.3",
    "eslint-plugin-jsdoc": "39.3.6",
    "eslint-plugin-jsx-a11y": "6.6.1",
    "eslint-plugin-no-null": "1.0.2",
    "eslint-plugin-prefer-arrow": "1.2.3",
    "eslint-plugin-prettier": "4.2.1",
    "eslint-plugin-react": "7.30.1",
    "eslint-plugin-react-hooks": "4.6.0",
    "eslint-plugin-simple-import-sort": "7.0.0",
    "eslint-plugin-unicorn": "43.0.2",
    "eslint-plugin-workspaces": "0.7.0",
    "faker": "4.1.0",
    "jest": "28.1.0",
    "jest-canvas-mock": "2.4.0",
    "jest-cli": "28.1.0",
    "jest-environment-jsdom": "28.1.0",
    "jest-fetch-mock": "3.0.3",
    "jsdom": "19.0.0",
    "jsdom-global": "3.0.2",
    "jsdom-worker": "0.1.0",
    "node-fetch": "2.6.0",
    "npm-run-all": "4.1.5",
    "prettier": "2.1.2",
    "rimraf": "3.0.0",
    "ts-jest": "28.0.1",
    "ts-node": "9.1.1",
    "typedoc": "0.15.3",
    "typescript": "4.6.4"
  },
  "jest-junit": {
    "suiteNameTemplate": "weavess-core/{filename}",
    "classNameTemplate": "weavess-core/{filename}/{classname}",
    "titleTemplate": "{title}"
  }
}
