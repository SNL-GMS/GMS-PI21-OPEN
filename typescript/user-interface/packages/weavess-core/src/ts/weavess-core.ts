import * as WeavessConstants from './constants';
import { Messages } from './messages';
import * as WeavessTypes from './types';
import * as WeavessUtil from './util';

export { WeavessConstants, Messages as WeavessMessages, WeavessTypes, WeavessUtil };
