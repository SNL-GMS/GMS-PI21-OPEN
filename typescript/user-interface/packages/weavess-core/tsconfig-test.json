{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "tsBuildInfoFile": "./.tsbuildinfo-test"
  },
  "references": []
}
