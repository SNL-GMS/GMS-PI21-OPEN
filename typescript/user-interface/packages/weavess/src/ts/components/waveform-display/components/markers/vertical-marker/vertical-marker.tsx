/* eslint-disable react/destructuring-assignment */
import { UILogger } from '@gms/ui-util';
import React from 'react';

import type { VerticalMarkerProps, VerticalMarkerState } from './types';

const logger = UILogger.create('GMS_LOG_WEAVESS', process.env.GMS_LOG_WEAVESS);

/**
 * VerticalMarker Component. Is not moveable
 */
export class VerticalMarker extends React.PureComponent<VerticalMarkerProps, VerticalMarkerState> {
  /** Ref to the marker container element */
  public containerRef: HTMLElement | null;

  public readonly lineBorderWidthPx: number = 1;

  /**
   * Constructor
   *
   * @param props Vertical Marker props as VerticalMarkerProps
   */
  public constructor(props: VerticalMarkerProps) {
    super(props);
    this.state = {};
  }

  // ******************************************
  // BEGIN REACT COMPONENT LIFECYCLE METHODS
  // ******************************************

  /**
   * Catches exceptions generated in descendant components.
   * Unhandled exceptions will cause the entire component tree to unmount.
   *
   * @param error the error that was caught
   * @param info the information about the error
   */
  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
  public componentDidCatch(error, info): void {
    logger.error(`Weavess Vertical Marker Error: ${error} : ${info}`);
  }

  // ******************************************
  // END REACT COMPONENT LIFECYCLE METHODS
  // ******************************************

  public render(): JSX.Element {
    return (
      <div
        className="vertical-marker"
        ref={ref => {
          this.containerRef = ref;
        }}
        key={this.props?.name}
        data-cy="weavess-static-vertical-marker"
        data-cy-left={`${this.props.percentageLocation}`}
        data-cy-color={`${this.props.color}`}
        style={{
          // offset the vertical marker by half its total width so it is centered.
          left: `calc(${this.props.percentageLocation}% - ${this.lineBorderWidthPx}px)`,
          border: `${this.lineBorderWidthPx}px ${this.props.lineStyle} ${this.props.color}`
        }}
      />
    );
  }
}
