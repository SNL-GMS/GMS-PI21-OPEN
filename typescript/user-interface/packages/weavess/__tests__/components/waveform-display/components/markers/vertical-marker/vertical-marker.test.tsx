/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable @typescript-eslint/no-magic-numbers */
import { WeavessTypes } from '@gms/weavess-core';
import { render } from '@testing-library/react';
import * as Enzyme from 'enzyme';
import * as React from 'react';

import type { VerticalMarkerProps } from '../../../../../../src/ts/components/waveform-display/components/markers/vertical-marker/types';
import { VerticalMarker } from '../../../../../../src/ts/components/waveform-display/components/markers/vertical-marker/vertical-marker';

const props: VerticalMarkerProps = {
  name: 'my-marker',
  color: '#ff0000',
  lineStyle: WeavessTypes.LineStyle.DASHED,
  percentageLocation: 90
};

const wrapper = Enzyme.mount(<VerticalMarker {...props} />);
const instance: VerticalMarker = wrapper.find(VerticalMarker).instance() as VerticalMarker;

describe('Weavess Vertical Marker', () => {
  it('Weavess Vertical Marker to be defined', () => {
    expect(VerticalMarker).toBeDefined();
  });

  it('renders', () => {
    const { container } = render(<VerticalMarker {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('componentDidCatch', () => {
    const spy = jest.spyOn(instance, 'componentDidCatch');
    instance.componentDidCatch(new Error('error'), { componentStack: undefined });
    expect(spy).toBeCalled();
  });
});
