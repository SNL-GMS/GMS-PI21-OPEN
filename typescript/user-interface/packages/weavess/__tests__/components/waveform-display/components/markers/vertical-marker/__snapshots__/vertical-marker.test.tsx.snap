// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weavess Vertical Marker renders 1`] = `
<div>
  <div
    class="vertical-marker"
    data-cy="weavess-static-vertical-marker"
    data-cy-color="#ff0000"
    data-cy-left="90"
    style="left: calc(90% - 1px); border: 1px dashed #ff0000;"
  />
</div>
`;
